pages/en/lb4/Routes.md

```diff
@@ -95,12 +94,11 @@ function greet(name: string) {
   return `hello ${name}`;
 }
 
-(async function start() {
-  const server = await app.getServer(RestServer);
-  const route = new Route('get', '/', spec, greet);
-  server.route(route);
-  await app.start();
-})();
+const app = new RestApplication();
+const route = new Route('get', '/', spec, greet);
+app.route(route); // attaches route to RestServer
+
+app.start();
```

Comment:
Isn't app.start() returning a promise and thus should be called with `await`? (see https://github.com/strongloop/loopback-next/blob/master/packages/core/src/application.ts#L128-L136)

Reply:
No, that's a myth.

1. `await` can only be used within an `async` function. The `await` keyword helps VM to suspend the execution and wait for the resolution/rejection of the promise or value.

2. In this case, you cannot use `await` as the module will be wrapped into a function by Node.js but it's not `async`.

3. The code before the change declares an `async` function and execute it immediately.

4. It's perfectly fine to call an `async` function or a function returning a promise without `await`. The return value is Promise. You have to manually call `.then() or .catch()` to set up result handler.

Reply:
Thanks for the clarification Raymond. We don't really need to show `.then() or .catch()` handlers in this example. Disregard my comment!